mne/beamformer/_lcmv.py

```diff
@@ -338,6 +338,53 @@ def apply_lcmv_epochs(epochs, filters, *, return_generator=False,
     return stcs
 
 
+@verbose
+def apply_lcmv_epochs_tfr(epochs_tfr, filters, *, return_generator=False,
+                          verbose=None):
+    """Apply Linearly Constrained Minimum Variance (LCMV) beamformer weights.
+
+    Apply Linearly Constrained Minimum Variance (LCMV) beamformer weights
+    on single trial time-frequency data.
+
+    Parameters
+    ----------
+    epochs_tfr : EpochsTFR
+        Time-frequency resolved epochs.
+    filters : list of Beamformer
+        LCMV spatial filter (beamformer weights)
+        Filter weights returned from :func:`make_lcmv` for each frequency.
+    return_generator : bool
+         Return a generator object instead of a list. This allows iterating
+         over the stcs without having to keep them all in memory.
+    %(verbose)s
+
+    Returns
+    -------
+    stcs: list | generator of (SourceEstimate | VolSourceEstimate)
+        The source estimates for all epochs.
+
+    See Also
+    --------
+    make_lcmv, apply_lcmv_epochs, apply_lcmv_raw, apply_lcmv, apply_lcmv_cov
+    """
+    info = epochs_tfr.info
+    tmin = epochs_tfr.times[0]
+
+    # TO DO: this is a very confusing list of lists and needs
+    # to be refactored into a class
+    stcs = list()
+    for freq_idx, these_filters in enumerate(filters):
+        sel = _check_channels_spatial_filter(
+            epochs_tfr.ch_names, these_filters)
+        data = epochs_tfr.data[:, sel, freq_idx, :]
```

Comment:
this only makes sense if the epochs_tfr.data are complex valued right? if it contains only the power in TF domain this is for me wrong.

Reply:
Why does this only work for complex-valued data and not power? I have not read up on the math but from seeing this presented and the name linearly-constrained minimum variance beamforming, I'm not sure why you couldn't use either...

Reply:
Because a spatial filter can only be applied on a linear time domain transform of the data eg Fourier. If you apply any non linearity to data eg to get power it is wrong

Reply:
Wow okay this is news to me! Can you also not do minimum norm estimation on power? And if you can, why is that different?